src/contract.py

```diff
@@ -184,8 +186,9 @@ def payoff(self, spot: dict[float, float]) -> float:
         if not set(timeline).issubset(set(spot.keys())):
             self.raise_missing_spot_error(list(spot.keys()))
         obs = [spot[t] for t in timeline]
-        in_out = self.barrier.in_out
-        is_breached = int(self.barrier.is_breached(obs))
+        in_out = self._barrier.get_in_out()
```

Comment:
Why access modifiers have been introduced again?

Reply:
They were not introduced, I have not yet deleted them. Will do it this week.